ifrs17/Report/ReportConfigurationAndUtils.ipynb

```diff
@@ -173,8 +173,7 @@
             "source": [
                 "public static async Task<ICollection<ReportVariable>> QueryReportVariablesAsync(this IWorkspace workspace, (int Year, int Month, string ReportingNode, string Scenario) args ) {",
                 "\n    ReportVariable[] reportVariables = new ReportVariable[0];",
-                "\n    if(args.Scenario != null) ",
-                "\n        reportVariables = (await workspace.QueryReportVariablesSingleScenarioAsync((args.Year, args.Month, args.ReportingNode, null)));",
+                "\n    if(args.Scenario != null) reportVariables = (await workspace.QueryReportVariablesSingleScenarioAsync((args.Year, args.Month, args.ReportingNode, null)));",
```

Comment:
IMO this whole method can be improved.

The query (await workspace.QueryReportVariablesSingleScenarioAsync((args.Year, args.Month, args.ReportingNode, args.Scenario) you always do in a way.

If args.Scenario == null, you can return immediately, I think.


I would say:
```
var bestEstimate = (await workspace.QueryReportVariablesSingleScenarioAsync((args.Year, args.Month, args.ReportingNode, null)));

return (args.Scenario != null) 
           ? bestEstimate
           : (await workspace.QueryReportVariablesSingleScenarioAsync((args.Year, args.Month, args.ReportingNode, args.Scenario))).Union(reportVariables.Select(x => x with {Scenario = args.Scenario}), EqualityComparer<ReportVariable>.Instance).ToArray();
```

Reply:
done